Index: src/drt_contact/contact_wear_lagrange_strategy.cpp
===================================================================
--- src/drt_contact/contact_wear_lagrange_strategy.cpp	(revision 21208)
+++ src/drt_contact/contact_wear_lagrange_strategy.cpp	(working copy)
@@ -33,7 +33,7 @@
 #include "../drt_io/io.H"
 
 #include "../linalg/linalg_utils.H"
-#include "../linalg/linalg_solver.H"
+#include "../linalg/linalg_multiply.H"
 #include <Epetra_FEVector.h>
 
 /*----------------------------------------------------------------------*
@@ -3929,6 +3929,9 @@
 *-----------------------------------------------------------------------*/
 void CONTACT::WearLagrangeStrategy::OutputWear()
 {
+#if 1
+  dserror("Linear solvers not included in contact module. Please contact AdCo Engineering GW GmbH for more details");
+#else
   //***********************************************
   //                 primvar wear
   //***********************************************
@@ -4099,6 +4102,7 @@
       }
     }
   }
+#endif /* if 0 */
   return;
 }