From b20b2e06e016fcb364b4c219e14f6c262566574c Mon Sep 17 00:00:00 2001
From: Tobias Wiesner <wiesner@lnm.mw.tum.de>
Date: Wed, 21 Jan 2015 11:44:41 +0100
Subject: [PATCH 15/16] Undo changes in SparseMatrix

---
 src/linalg/linalg_sparsematrix.H |   13 -------------
 1 files changed, 0 insertions(+), 13 deletions(-)

diff --git a/src/linalg/linalg_sparsematrix.H b/src/linalg/linalg_sparsematrix.H
index a7ecc9e..119ce0b 100644
--- a/src/linalg/linalg_sparsematrix.H
+++ b/src/linalg/linalg_sparsematrix.H
@@ -84,19 +84,6 @@ class SparseMatrix : public SparseMatrixBase
      */
     enum MatrixType {CRS_MATRIX, FE_MATRIX};
 
-    /// The following dummy templated constructors will catch all attempts to call any constructor with arguments
-    /// that require implicit casting. Therefore you have to instantiate this class with exactly the types
-    /// matching one of the constructors
-#if __cplusplus >= 201103L
-  template <typename... T> foo(T...) = delete;
-#else
-  template <typename T0>                                                     SparseMatrix(T0)                 {BOOST_STATIC_ASSERT_MSG(sizeof(T0) == 0, "dummy templated constructor") __attribute__((unused));}
-  template <typename T0, typename T1>                                        SparseMatrix(T0, T1)             {BOOST_STATIC_ASSERT_MSG(sizeof(T0) == 0, "dummy templated constructor") __attribute__((unused));}
-  template <typename T0, typename T1, typename T2>                           SparseMatrix(T0, T1, T2)         {BOOST_STATIC_ASSERT_MSG(sizeof(T0) == 0, "dummy templated constructor") __attribute__((unused));}
-  template <typename T0, typename T1, typename T2, typename T3>              SparseMatrix(T0, T1, T2, T3)     {BOOST_STATIC_ASSERT_MSG(sizeof(T0) == 0, "dummy templated constructor") __attribute__((unused));}
-  template <typename T0, typename T1, typename T2, typename T3, typename T4> SparseMatrix(T0, T1, T2, T3, T4) {BOOST_STATIC_ASSERT_MSG(sizeof(T0) == 0, "dummy templated constructor") __attribute__((unused));}
-#endif
-
     /// construction of sparse matrix
     SparseMatrix(
       Teuchos::RCP<Epetra_CrsGraph> crsgraph,
-- 
1.7.7.6

